src/components/SetupTab/index.jsx

```diff
@@ -83,18 +82,22 @@ function initializeArgValues(argsSpec, argsDict) {
       touched: !initIsEmpty, // touch them only if initializing with values
     };
   });
-  return ({ argsValues: argsValues, argsValidation: argsValidation });
+  return ({ argsValues: argsValues });
 }
 
 /** Renders an arguments form, execute button, and save buttons. */
 export default class SetupTab extends React.Component {
   constructor(props) {
     super(props);
+    // map each arg to an empty object, to fill in later
+    const argsValidation = Object.keys(props.argsSpec).reduce((acc, argkey) => {
+      acc[argkey] = {};
+      return acc;
+    }, {});
     this.state = {
       argsValues: null,
-      argsValidation: {},
-      argsValid: false,
-      sortedArgKeys: null,

```

Comment:
since this is now just `this.props.uiSpec.order` I don't think it needs to be part of the `state` anymore